Adyen/Core/AdyenContext/AdyenContext.swift

```diff
@@ -24,17 +27,28 @@ public final class AdyenContext {
     /// - Parameters:
     ///   - apiContext: The API context used to retrieve internal resources.
     ///   - analyticsConfiguration: A configuration object that specifies the behavior for the analytics.
-    public init(apiContext: APIContext, analyticsConfiguration: AnalyticsConfiguration = .init()) {
+    ///   - payment: The payment information.
+    public init(apiContext: APIContext, payment: Payment?, analyticsConfiguration: AnalyticsConfiguration = .init()) {
         self.apiContext = apiContext
+        self.payment = payment
 
         let apiClient = APIClient(apiContext: apiContext)
         self.analyticsProvider = AnalyticsProvider(apiClient: apiClient,
                                                    configuration: analyticsConfiguration)
     }
 
     internal init(apiContext: APIContext,
+                  payment: Payment?,
                   analyticsProvider: AnalyticsProviderProtocol) {
         self.apiContext = apiContext
         self.analyticsProvider = analyticsProvider
+        self.payment = payment
+    }
+
+    @_spi(AdyenInternal)
+    public func updated(with payment: Payment?) -> AdyenContext {
```

Comment:
Maybe similar to others, `replacingPayment(with payment: Payment?)`

Reply:
How about:

```swift
replacing(payment: Payment?)
```